src/net/udp.rs

```diff
@@ -183,19 +186,22 @@ impl UdpSocket {
         Self { inner }
     }
 
-    /// Connects this UDP socket to a remote address, allowing the `write` and
-    /// `read` syscalls to be used to send data and also applies filters to only
-    /// receive data from the specified address.
+    /// "Connects" this UDP socket to a remote address.
+    ///
+    /// This enables `write` and `read` syscalls to be used on this instance.
+    /// It also constrains the `read` to receive data only from the specified remote peer.
     ///
-    /// Note that usually, a successful `connect` call does not specify
-    /// that there is a remote server listening on the port, rather, such an
-    /// error would only be detected after the first send.
+    /// Note: UDP is connectionless, so a successful `connect` call does not execute
+    /// a handshake or validation of the remote peer of any kind.
+    /// Any errors would not be detected until the first send.
     pub async fn connect(&self, socket_addr: SocketAddr) -> io::Result<()> {
         self.inner.connect(SockAddr::from(socket_addr)).await
     }
 
-    /// Sends data on the socket to the given address. On success, returns the
-    /// number of bytes written.
+    /// Sends data on the socket to the given address.
+    ///
+    /// On success, returns the number of bytes written.
+    /// This can be used on a socket regardless of whether it has been "connected".
```

Comment:
This aside, line 204, isn't really necessary. The `connect` method already calls out what it enables. Everything else is normal for a UDP socket.

Reply:
Agreed.